pkg/kubectl/resource/builder.go

```diff
@@ -757,3 +774,17 @@ func HasNames(args []string) (bool, error) {
 	}
 	return hasCombinedTypes || len(args) > 1, nil
 }
+
+func (b *Builder) validateNamespace(c RESTClient, namespace string) error {
+	if !b.validateNamespaces {
+		return nil
+	}
+	if namespace == api.NamespaceDefault || namespace == api.NamespaceAll {
+		return nil
+	}
+	err := c.Get().AbsPath("api", "v1", "namespaces", namespace).Do().Error()
```

Comment:
I don't think we can do that in this case.  I looked into it, but the truth is that at the registry level nothing really knows anything about namespaces, they're just full path keys.  And also, none of the registries know about the namespace registry.

So in order to accomplish this we would need to both:
   * Make the generic registry aware of namespaces
   * Plumb the namespace registry down into all of the other registries so that they could do the namespace lookup.

I don't think that degree of surgery is worth it.  I'm also not clear that web dashboards will ever encounter this problem, since they tend not to allow people to specify arbitrary namespaces, instead preferring to list existing namespaces and allow the user to select via GUI.